schema/validator.go

```diff
@@ -88,11 +89,43 @@ func (v Validator) Validate(src io.Reader) error {
 
 type unimplemented string
 
-func (v unimplemented) Validate(src io.Reader) error {
+func (v unimplemented) Validate(src io.Reader, strict bool) error {
 	return fmt.Errorf("%s: unimplemented", v)
 }
 
-func validateManifestDescendants(r io.Reader) error {
+func validateMediaTypeInManifestList(r io.Reader, strict bool) error {
+	header := v1.ManifestList{}
+
+	buf, err := ioutil.ReadAll(r)
+	if err != nil {
+		return errors.Wrapf(err, "error reading the io stream")
+	}
+
+	err = json.Unmarshal(buf, &header)
+	if err != nil {
+		return errors.Wrap(err, "manifest list format mismatch")
+	}
+
+	if header.MediaType != string(v1.MediaTypeImageManifestList) {
+		msg := fmt.Sprintf("manifest list has an unknown media type: %s", header.MediaType)
+		if strict {
+			return fmt.Errorf("error: %s", msg)
```

Comment:
I'm fine with the “warning: ” prefix you have below, but I don't think you need an “error: ” prefix for the error string.  You know it's an error because its Go type is `error`, having “error” in the string payload is redundant.  If you want to display these to users with an error prefix (which is fine with me), then prefix the error with “error: ” when you print it to stderr (or wherever).  But don't add the prefix here.